server/openai/test.js

```diff
@@ -0,0 +1,30 @@
+const generateItinerary = require('./generateItenerary')
+const Itinerary = require('../models/itineraryModel')
+
+const exampleItinerary = new Itinerary({
+  destination: 'Paris',
+  budget: 2000,
+  days: 5,
+  stages: 3,
+  preferences: ['shopping', 'museums', 'historical landmarks'],
+})
+
+;(async function testGenerateItinerary() {
```

Comment:
curious about this notation with a semicolon at the front, what does it do?

Reply:
defining and running the function at the same time. GPT response:
```
This is an example of an immediately invoked async function. It is defined and executed immediately after its declaration. This pattern is useful when you need to execute asynchronous code in places where you can't use the "await" keyword directly, such as the global scope or inside non-async functions.
```

Reply:
oh that's cool! thnx